auth/install/index.php

```diff
@@ -109,94 +108,100 @@
-		$link = mysqli_connect("$database_host", "$database_username", "$database_password");
-		if (!$link) {
-			$errorMessage[] = "Could not connect to the server '" . $database_host . "'<br />MySQL Error: " . mysqli_error($link);
-		}else{
+	if (!$database->acquire('dbuser'))
+		$errorMessage[] = 'User name is required';
 
-			//next check that the database exists
-			$dbcheck = mysqli_select_db($link, "$database_name");
-			if (!$dbcheck) {
-				$errorMessage[] = "Unable to access the database '" . $database_name . "'.  Please verify it has been created.<br />MySQL Error: " . mysqli_error($link);
-			}else{
-				//passed db host, name check, test that user can select from Auth database
-				$result = mysqli_query($link, "SELECT loginID FROM " . $database_name . ".User WHERE loginID like '%coral%';");
-				if (!$result){
-					$errorMessage[] = "Unable to select from the User table in database '" . $database_name . "' with user '" . $database_username . "'.  Error: " . mysqli_error($link);
-				}
+	if (!$database->acquire('dbpass'))
+		$errorMessage[] = 'Password is required';
 
-			}
-		}
+	if (!$admin->acquire('coral_username'))
+		$errorMessage[] = 'CORAL Admin Username is required';
+	if (!$admin->acquire('coral_password'))
+		$errorMessage[] = 'CORAL Admin Password is required';
 
-	}
+	$passwordPrefix = Util::randomString(45);
+	$password 		= Util::hashString('sha512', $passwordPrefix . $admin->coral_password);
+	$create_admin_query = "INSERT INTO `User` VALUES ('$admin->coral_username','$password','$passwordPrefix','Y')";
 
+	if (!$general->acquire('session_timeout'))
+		$errorMessage[] = 'Session timeout is required';
+
+	$ldap->acquire('ldap_enabled',false,true);
+	$ldap->acquire('ldap_host');
+	$ldap->acquire('ldap_port');
+	$ldap->acquire('ldap_search_key');
+	$ldap->acquire('ldap_base_dn');
+	$ldap->acquire('ldap_bind_account');
+	$ldap->acquire('ldap_bind_password');
+
+    if ($ldap->ldap_enabled) {
+        if (!$ldap->ldap_host)       $errorMessage[] = "LDAP Host is required for LDAP";
+        if (!$ldap->ldap_search_key) $errorMessage[] = "LDAP Search Key is required for LDAP";
+        if (!$ldap->ldap_base_dn)    $errorMessage[] = "LDAP Base DN is required for LDAP";
+    }
+
+	//only continue to checking DB connections if there were no errors this far
+	if (count($errorMessage) > 0){
+		$general->step="3";
+
+	//first check connecting to host
+	} else if ($errmsg = $db->connect("$database->dbhost", "$database->dbuser", "$database->dbpass")) {
+		$errorMessage[] = $errmsg;
+
+	//next check that the database exists
+	} else if ($errmsg = $db->selectDB("$database->dbname")) {
+		$errorMessage[] = $errmsg;
+	} else if (!$db->query($create_admin_query)) {
+		$errorMessage[] = "Failed to create CORAL Admin";
+	//passed db host, name check, test that user can select from Auth database
+	} else if (!$db->query("SELECT loginID FROM $database->dbname.User WHERE loginID like '%$admin->coral_username%';")){

```

Comment:
This line is hard to read. Could the variables be separated from the query string to improve readability? The result would be something like this:
...$db->query("SELECT loginID FROM " . $database->dbname . ".User WHERE loginID like '%" . $admin->coral_username . "%';")...